examples/build.gradle

```diff
@@ -0,0 +1,123 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.tools.ant.taskdefs.condition.Os
+
+apply plugin: 'java'
+
+description = 'Ballerina - MQTT Examples'
+
+def ballerinaDist = "${project.rootDir}/target/ballerina-runtime"
+def examples = ["temperature-sensor/temperature-publisher", "temperature-sensor/temperature-analyzer"]
+def packageName = "mqtt"
+
+dependencies {
+    implementation group: 'org.ballerinalang', name: 'ballerina-lang', version: "${ballerinaLangVersion}"
+    implementation group: 'org.ballerinalang', name: 'ballerina-tools-api', version: "${ballerinaLangVersion}"
+    implementation group: 'org.ballerinalang', name: 'ballerina-parser', version: "${ballerinaLangVersion}"
+    implementation group: 'org.testng', name: 'testng', version: "${testngVersion}"
+}
+
+clean {
+    examples.forEach { example ->
+        delete "${projectDir}/${example}/target"
+    }
+}
+
+task testExamples {
+    doLast {
+        examples.each { example ->
+            try {
+                exec {
+                    workingDir project.projectDir
+                    if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                        commandLine 'cmd', '/c', "${ballerinaDist}/bin/bal.bat test --offline ${example} && exit %%ERRORLEVEL%%"
+                    } else {
+                        commandLine 'sh', '-c', "${ballerinaDist}/bin/bal test --offline ${example}"
+                    }
+                }
+            } catch (Exception e) {
+                println("Example '${example}' Build failed: " + e.message)
+                throw e
+            }
+        }
+    }
+}
+
+task buildExamples {
+    gradle.taskGraph.whenReady { graph ->
+        if (graph.hasTask(":${packageName}-examples:test")) {
+            buildExamples.enabled = false
+        } else {
+            testExamples.enabled = false
+        }
+    }
+    doLast {
+        examples.each { example ->
+            try {
+                exec {
+                    workingDir project.projectDir
+                    if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                        commandLine 'cmd', '/c', "${ballerinaDist}/bin/bal.bat build ${example} && exit %%ERRORLEVEL%%"
+                    } else {
+                        commandLine 'sh', '-c', "${ballerinaDist}/bin/bal build ${example}"
+                    }
+                }
+            } catch (Exception e) {
+                println("Example '${example}' Build failed: " + e.message)
+                throw e
+            }
+        }
+    }
+}
+
+task stopMqttServer() {
```

Comment:
What happens if someone exclude the `mqtt-examples` from the gradle build using `-x` flag ? Will the `mqtt-server` stop properly ?

Reply:
Since there are no tests, don't need this at the moment. Reverted to having in ballerina/build.gradle